### PR TITLE
Use ansibullbot._json_compat.json_dumps everywhere

### PR DESCRIPTION
**Pull request by pilou-**
See https://github.com/ansible/ansibullbot/pull/1117#discussion_r240239224. I am not sure why `json_dumps` wasn't already used everywhere ?

cc @webknjaz

### CODE REVIEW

**Comment by webknjaz** [comment]
I probably just missed some places.

**Comment by webknjaz** [comment]
@abadger plz take a look as well

**Comment by abadger** [comment]
I don't think this is quite correct and looking at the iterencode() method... it still looks wrong to me.

* For this PR, I think you need to use to_bytes around most of the json.dumps() calls.  I'll make a suggestion for one of the calls so that you can see what I mean

* For iterencode... why is the text transformation dependent on whether ensure_ascii is set?  IIRC, ensure_ascii means that non-ascii will be escaped in the output.  The input can contain non-ascii which should be text type in either case.

File: ansibullbot/triagers/ansible.py [hunk 1]
```diff
@@ -1545,7 +1545,7 @@ def move_issue(self, issue):
         # cache the data in case of errors
         mdata = self.IM.migration_map.get(issue.html_url)
         with open(mfile, 'wb') as f:
-            f.write(json.dumps(mdata, indent=2))
+            f.write(json_dumps(mdata))
```
Comment (abadger) [inline ansibullbot/triagers/ansible.py]:
```suggestion
            f.write(to_bytes(json_dumps(mdata), errors='surrogate_or_strict'))
```

The file is opened for writing in binary mode.  So you always want to write bytes to the file, not text.

Could you do this the other way and open the file in non-binary mode?  That won't work on python2 as open in python2 always expects bytes.  So you can get UnicodeError tracebacks if you write text to the filehandle on python2 when python2 implicitly tries to convert from text to bytes.

File: ansibullbot/utils/gh_gql_client.py [hunk 2]
```diff
@@ -227,7 +227,7 @@ def get_summaries(self, owner, repo, otype='issues', last=None, first='first: 10
                 u'variables': u'{}',
                 u'operationName': None
             }
-            rr = requests.post(self.baseurl, headers=self.headers, data=json.dumps(payload))
+            rr = requests.post(self.baseurl, headers=self.headers, data=json_dumps(payload))
```
Comment (abadger) [inline ansibullbot/utils/gh_gql_client.py]: From this old bug: https://github.com/requests/requests/issues/3875 you may have to use to_bytes() with this as well.  I'm not certain, though.

File: ansibullbot/utils/shippable_api.py [hunk 3]
```diff
@@ -138,7 +139,7 @@ def _load_cache_file(self, cfile):
 
     def _write_cache_file(self, cfile, data):
         with gzip.open(cfile, 'w') as f:
```
Comment (abadger) [inline ansibullbot/utils/shippable_api.py]: This should be:
```suggestion
        with gzip.open(cfile, 'wb') as f:
```

File: ansibullbot/wrappers/defaultwrapper.py [hunk 4]
```diff
@@ -1016,7 +1017,7 @@ def log_ci_status(self, status_data):
                 jdata[turl][u'history'][ts] = sd[u'state']
 
         with open(logfile, 'w') as f:
```
Comment (abadger) [inline ansibullbot/wrappers/defaultwrapper.py]:
```suggestion
        with open(logfile, 'wb') as f:
```

File: ansibullbot/utils/gh_gql_client.py (hunk 2 above)
Comment (webknjaz) [reply ansibullbot/utils/gh_gql_client.py]: Well, requests methods support `json=` arg so it's better just use it then:
```suggestion
            rr = requests.post(self.baseurl, headers=self.headers, json=payload)
```

File: ansibullbot/triagers/ansible.py [hunk 5]
```diff
@@ -2450,4 +2450,4 @@ def set_resume(self, repo, number):
         }
         resume_file = os.path.join(self.cachedir_base, u'resume.json')
         with open(resume_file, 'wb') as f:
-            f.write(json.dumps(data, indent=2))
+            f.write(json_dumps(data))
```
Comment (webknjaz) [inline ansibullbot/triagers/ansible.py]: same as above I guess

Comment (abadger) [reply ansibullbot/triagers/ansible.py]: Yep, all the f.write's should do to_bytes() on the data.

File: ansibullbot/utils/gh_gql_client.py (hunk 2 above)
Comment (abadger) [reply ansibullbot/utils/gh_gql_client.py]: Sounds good to me.

**Comment by webknjaz** [comment]
@pilou- would you mind addressing the comments, please?

**Comment by mkrizek** [comment]
I am going to go ahead and close this due to inactivity and because there is a plan to remove Python 2 support from the bot anyway.

Thanks @pilou- !

**Pull request closed without merging**